core/core/test/serverspec/spec/master/master_spec.rb

```diff
@@ -1,28 +1,158 @@
+    its(:stdout) { should eq count_inventory_roles("master").to_s }
+  end
+end
+
+describe 'Check if the number of worker nodes is the same as indicated in the inventory file' do
+  describe command('kubectl get nodes --no-headers | grep -v master | wc -l | tr -d "\n"') do
+    its(:stdout) { should eq count_inventory_roles("worker").to_s }
+  end
+end
+
+describe 'Check if there are any nodes that have status other than Ready' do
+  describe command('kubectl get nodes') do
+    its(:stdout) { should match /Ready/ }
+    its(:stdout) { should_not match /NotReady/ }
+    its(:stdout) { should_not match /Unknown/ }
+    its(:stdout) { should_not match /False/ }
+  end
+end  
 
-describe port(6443) do
+describe 'Check the port on which to serve HTTPS with authentication and authorization' do
+  describe port(6443) do
+    let(:disable_sudo) { false }
     it { should be_listening }
   end
+end  
+
+describe 'Creating, checking and decoding a secret using kubectl' do
+
+  test_user = 'user123'
+  test_pass = 'pass456'
+  test_user_b64 = Base64.encode64(test_user)
+  test_pass_b64 = Base64.encode64(test_pass)
+  timestamp = DateTime.now.to_time.to_i.to_s
+  test_secret = 'testsecret' + timestamp
+
+  describe 'Creating a secret...' do
+    describe command("kubectl create secret generic #{test_secret} --from-literal=username=#{test_user} --from-literal=password=#{test_pass}") do
+      its(:stdout) { should match /secret\/#{test_secret} created/ }
+    end
+  end
+  describe 'Checking if it exists...' do
+    describe command("kubectl get secrets") do
+      its(:stdout) { should match /#{test_secret}/ }
+    end
+  end
+  describe 'Decoding...' do
+    describe command("kubectl get secret #{test_secret} -o yaml") do
+      its(:stdout) { should match /name: #{test_secret}/ }
+      its(:stdout) { should match /#{test_user_b64}/ }
+      its(:stdout) { should match /#{test_pass_b64}/ }
+    end
+  end
+end  
+
+describe 'Testing kubernetes dashboard...' do
+  describe 'Looking for kubernetes-dashboard pod' do
+    describe command('kubectl get pods --all-namespaces | grep kubernetes-dashboard') do
+        its(:stdout) { should match /kubernetes-dashboard/ }
+        its(:stdout) { should match /Running/ }
+        its(:exit_status) { should eq 0 }
+    end
+  end
+  describe 'Checking kubernetes-dashboard deployment status' do
+    describe command('kubectl get deployments --all-namespaces --field-selector metadata.name=kubernetes-dashboard') do
+      its(:stdout) { should match /kubernetes-dashboard/ }
+      its(:stdout) { should match /1\/1/ }
+      its(:exit_status) { should eq 0 }
+    end
+  end
+  describe 'Getting admin token bearer' do
+    describe command("kubectl describe secret $(kubectl get secrets --namespace=kube-system | grep admin-user \
+    | awk '{print $1}') --namespace=kube-system | grep -E '^token' | awk '{print $2}' | head -1") do
```

Comment:
for future reference, this: `grep -E '^something' | awk '{print $1}'`
can be replaced with single awk:  `awk '/^something/ {print $1}'`

Reply:
fixed in 5683acf